### ISSUE TITLE
Problem with Z.EntityFramework.Plus Generated Query not following Ef Core Relational Names

### ISSUE THREAD

**Issue by tdolgyras** [zzzprojects/EntityFramework-Plus#487]
### Description
We use multiple Database Providers in our project. Our naming convetions in our c# code is PascalCase. In PostgreSql, we have a fix setiing PascalCase in lower_case_with_underscores. So, we can have easier sql code in PostgreSql.

### Exception
When we execute this code:
`await _db.PostsHistories.Where(xsh => xsh.PostId == xr.Id).DeleteAsync();`
We get a PostgreSql exception containing wrong sql in PascalCase.
In Z version 1.8.18 it was working fine.

### Further technical details
- EF version: Microsoft.EntityFrameworkCore 2.2.4
- Z version: Z.EntityFramework.Plus.EFCore 1.8.27
- Database Provider: MsSql, Pomelo.EntityFrameworkCore.MySql, Npgsql.EntityFrameworkCore.PostgreSQL

Our PostgreSQL code
```
 protected override void OnModelCreating(ModelBuilder modelBuilder)
{
            base.OnModelCreating(modelBuilder);
            FixSnakeCaseNames(modelBuilder);
}
#region Fix PostgreSql Naming
private static readonly Regex _keysRegex = new Regex("^(PK|FK|IX)_", RegexOptions.Compiled);
private void FixSnakeCaseNames(ModelBuilder modelBuilder)
{
	var mapper = new NpgsqlSnakeCaseNameTranslator();
	foreach (var table in modelBuilder.Model.GetEntityTypes())
	{
		ConvertToSnake(mapper, table);
		foreach (var property in table.GetProperties())
		{
			ConvertToSnake(mapper, property);
		}

		foreach (var primaryKey in table.GetKeys())
		{
			ConvertToSnake(mapper, primaryKey);
		}

		foreach (var foreignKey in table.GetForeignKeys())
		{
			ConvertToSnake(mapper, foreignKey);
		}

		foreach (var indexKey in table.GetIndexes())
		{
			ConvertToSnake(mapper, indexKey);
		}
	}
}

private void ConvertToSnake(INpgsqlNameTranslator mapper, object entity)
{
	switch (entity)
	{
		case IMutableEntityType table:
			var relationalTable = table.Relational();
			relationalTable.TableName = ConvertGeneralToSnake(mapper, relationalTable.TableName);
			//if (relationalTable.TableName.StartsWith("asp_net_"))
			//{
			//    relationalTable.TableName = relationalTable.TableName.Replace("asp_net_", string.Empty);
			//    relationalTable.Schema = "identity";
			//}

			break;
		case IMutableProperty property:
			property.Relational().ColumnName = ConvertGeneralToSnake(mapper, property.Relational().ColumnName);
			break;
		case IMutableKey primaryKey:
			primaryKey.Relational().Name = ConvertKeyToSnake(mapper, primaryKey.Relational().Name);
			break;
		case IMutableForeignKey foreignKey:
			foreignKey.Relational().Name = ConvertKeyToSnake(mapper, foreignKey.Relational().Name);
			break;
		case IMutableIndex indexKey:
			indexKey.Relational().Name = ConvertKeyToSnake(mapper, indexKey.Relational().Name);
			break;
		default:
			throw new NotImplementedException("Unexpected type was provided to snake case converter");
	}
}

private string ConvertKeyToSnake(INpgsqlNameTranslator mapper, string keyName) =>
	ConvertGeneralToSnake(mapper, _keysRegex.Replace(keyName, match => match.Value.ToLower()));

private string ConvertGeneralToSnake(INpgsqlNameTranslator mapper, string entityName) =>
	mapper.TranslateMemberName(ModifyNameBeforeConvertion(mapper, entityName));

protected virtual string ModifyNameBeforeConvertion(INpgsqlNameTranslator mapper, string entityName) => entityName;

#endregion
```

**Comment by JonathanMagnan** [comment]
Hello @tdolgyras ,

Could you provide a runnable project/solution with this issue? It will help my developer to get started with it.

Best Regards,

Jonathan

**Comment by xrkolovos** [comment]
This stooped working in version 1.8.20-1.8.29. It was working until 1.8.19.

[Z.EntityFramework.Plus.Issue487.zip](https://github.com/zzzprojects/EntityFramework-Plus/files/3231180/Z.EntityFramework.Plus.Issue487.zip)

**Comment by JonathanMagnan** [comment]
Thank you,

My developer made a fix yesterday for an issue probably similar to your.

We will test the fix later and probably release it today. I will ask him to test with your project as well first.

**Comment by JonathanMagnan** [comment]
Hello @xrkolovos 

A new version has been released v1.9.0, it's a hybrid of your fix.

From our test, we found out that sometimes the relational name is used and sometimes the property name is used (via an alias) depending on the number of the primary key of your entities.

I'm not sure if we are missing some case but it fixes your project.

Let me know if the latest version works correctly.

**Comment by xrkolovos** [comment]
It worked perfectly.
